programs/spammers.go

```diff
@@ -11,89 +10,144 @@ import (
-			log.Debugf("Prepared %d unspent outputs for spamming.", unspentOutputsLeft)
-		}
+		case <-time.After(checkFundsLeftInterval):
+			outputsLeft := w.UnspentOutputsLeft(evilwallet.Fresh)
+			// less than one big wallet left
+			if outputsLeft < evilwallet.FaucetRequestSplitNumber*evilwallet.FaucetRequestSplitNumber {
+				log.Debugf("Requesting new faucet funds, outputs left: %d", outputsLeft)
+				err := w.RequestFreshBigFaucetWallet()
+				if err != nil {
+					log.Errorf("Failed to request faucet wallet: %s, stopping next requests...", err)
 
-		switch sType {
-		case spammer.TypeBlock:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-				s := SpamBlocks(w, params.Rates[i], params.TimeUnit, params.Durations[i], params.BlkToBeSent[i], params.EnableRateSetter, params.AccountAlias)
-				if s == nil {
-					return
-				}
-				s.Spam()
-			}(i)
-		case spammer.TypeBlowball:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-
-				s := SpamBlowball(w, params.Rates[i], params.TimeUnit, params.Durations[i], params.BlowballSize, params.EnableRateSetter, params.AccountAlias)
-				if s == nil {
 					return
 				}
-				s.Spam()
-			}(i)
-		case spammer.TypeTx:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-				SpamTransaction(w, params.Rates[i], params.TimeUnit, params.Durations[i], params.DeepSpam, params.EnableRateSetter, params.AccountAlias)
-			}(i)
-		case spammer.TypeDs:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-				SpamDoubleSpends(w, params.Rates[i], params.NSpend, params.TimeUnit, params.Durations[i], params.DelayBetweenConflicts, params.DeepSpam, params.EnableRateSetter, params.AccountAlias)
-			}(i)
-		case spammer.TypeCustom:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-				s := SpamNestedConflicts(w, params.Rates[i], params.TimeUnit, params.Durations[i], params.Scenario, params.DeepSpam, false, params.EnableRateSetter, params.AccountAlias)
-				if s == nil {
-					return
-				}
-				s.Spam()
-			}(i)
-		case spammer.TypeAccounts:
-			wg.Add(1)
-			go func(i int) {
-				defer wg.Done()
-
-				s := SpamAccounts(w, params.Rates[i], params.TimeUnit, params.Durations[i], params.EnableRateSetter, params.AccountAlias)
-				if s == nil {
-					return
-				}
-				s.Spam()
-			}(i)
 
-		default:
-			log.Warn("Spamming type not recognized. Try one of following: tx, ds, blk, custom, commitments")
+				log.Debugf("Requesting finished, currently available: %d unspent outputs for spamming.", w.UnspentOutputsLeft(evilwallet.Fresh))
+			}
 		}
 	}
+}
+
+func CustomSpam(params *CustomSpamParams, accWallet *accountwallet.AccountWallet) {
+	w := evilwallet.NewEvilWallet(evilwallet.WithClients(params.ClientURLs...), evilwallet.WithAccountsWallet(accWallet))
+	wg := sync.WaitGroup{}
+
+	log.Infof("Start spamming with rate: %d, time unit: %s, and spamming type: %s.", params.Rate, params.TimeUnit.String(), params.SpamType)
+
+	var duration time.Duration = -1
+	if params.Duration >= 0 {
```

Comment:
Isn't this always true?

Reply:
I set `params.Duration=-1` during the prameters parsing, so by checking if time.Duration >=  I check if it's not an infinite spam

Reply:
I replaced all -1 with const